File: jar-auth/src/test/java/com/beancounter/auth/RealmTest.java
package com.beancounter.auth;

import static org.assertj.core.api.Assertions.assertThat;

import com.beancounter.auth.common.TokenUtils;
import com.beancounter.auth.server.JwtRoleConverter;
import com.beancounter.common.model.SystemUser;
import com.beancounter.common.utils.KeyGenUtils;
import java.util.UUID;
import org.junit.jupiter.api.Test;
import org.springframework.security.core.authority.SimpleGrantedAuthority;

public class RealmTest {
  @Test
  void is_RoleConverterValid() {
    JwtRoleConverter jwtRoleConverter =
        new JwtRoleConverter("empty", "nothing");

    assertThat(jwtRoleConverter.getAuthorities(
        TokenUtils.getUserToken(
            SystemUser.builder()
                .id(KeyGenUtils.format(UUID.randomUUID()))
                .build())))
        .hasSize(3)
        .containsExactlyInAnyOrder(
            // Default scopes
            new SimpleGrantedAuthority("SCOPE_email"),
            new SimpleGrantedAuthority("SCOPE_profile"),
            new SimpleGrantedAuthority("SCOPE_beancounter"));
  }
}

File: jar-auth/src/test/java/com/beancounter/auth/RealmTest.kt
package com.beancounter.auth

import com.beancounter.auth.common.TokenUtils
import com.beancounter.auth.server.JwtRoleConverter
import com.beancounter.common.model.SystemUser
import com.beancounter.common.utils.KeyGenUtils
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.security.core.authority.SimpleGrantedAuthority
import java.util.UUID

/**
 * Converts OAuth realms to Spring Authorities
 */
class RealmTest {
    @Test
    fun roleConverterdFromOAuthToGrantedAuth() {
        val jwtRoleConverter = JwtRoleConverter("empty", "nothing")
        val su = SystemUser(KeyGenUtils().format(UUID.randomUUID()))
        Assertions.assertThat(
            jwtRoleConverter.getAuthorities(
                TokenUtils().getUserToken(su)
            )
        )
            .hasSize(3)
            .containsExactlyInAnyOrder( // Default scopes
                SimpleGrantedAuthority("SCOPE_email"),
                SimpleGrantedAuthority("SCOPE_profile"),
                SimpleGrantedAuthority("SCOPE_beancounter")
            )
    }
}
